cg/services/deliver_files/deliver_files_service/deliver_files_service_factory.py

```diff
@@ -102,12 +82,13 @@ def _get_file_fetcher(self, delivery_type: DataDelivery) -> FetchDeliveryFilesSe
             tags_fetcher=file_tag_fetcher,
         )
 
-    @staticmethod
     def _get_sample_file_formatter(
-        workflow: Workflow,
+        self,
+        case: Case,
     ) -> SampleFileFormatter | SampleFileConcatenationFormatter:
         """Get the file formatter service based on the workflow."""
-        if workflow in [Workflow.MICROSALT]:
+        converted_workflow: Workflow = self._convert_workflow(case)
+        if converted_workflow in [Workflow.MICROSALT]:
             return SampleFileConcatenationFormatter(FastqConcatenationService())
         return SampleFileFormatter()
```

Comment:
```suggestion
        needs_fastq_concatenation: bool = self._needs_fastq_concatenation(case)
        if needs_fastq_concatenation:
            return SampleFileConcatenationFormatter(FastqConcatenationService())
        return SampleFileFormatter()
```

Reply:
This is a valid point that we discussed with @ChrOertlin. We agreed that the changes that Chris and Eliot are working on regarding Mutant will kind of override this, as more logic will be introduced regarding when to concatenate. We decided to leave it as "converted workflow"